haystack_experimental/components/tools/openapi/types.py

```diff
@@ -37,9 +38,41 @@ def path_to_operation_id(path: str, http_method: str = "get") -> str:
     :param http_method: The HTTP method to use for the operationId.
     :returns: The operationId.
     """
-    if http_method.lower() not in VALID_HTTP_METHODS:
-        raise ValueError(f"Invalid HTTP method: {http_method}")
-    return path.replace("/", "_").lstrip("_").rstrip("_") + "_" + http_method.lower()
+    return sanitize_function_name(path + "_" + http_method.lower())
+
+
+def sanitize(data: Dict[str, Any]) -> Dict[str, Any]:
+    """
+    Sanitizes the given function calling definition by adjusting its properties to LLM requirements.
```

Comment:
`LLM requirements` is rather vague. Is there some documentation that defines what that means, i.e., where are these requirements coming from?

Reply:
@shadeMe I couldn't find this in documentation but you hit it in the error reports returned by LLM once you use enough openapi specs. For example:
```
# BadRequestError: Error code: 400 - {'error': {'message': "Invalid 'tools[0].function.name': string does not match pattern. Expected a string that matches the pattern '^[a-zA-Z0-9_-]+$'.", 'type': 'invalid_request_error', 'param': 'tools[0].function.name', 'code': 'invalid_value'}}
```

Once you apply this pattern to method is start to work. Same for length of description field

Reply:
Which LLM? Which provider? Are there providers/models that don't accept underscores? If this code doesn't follow an official spec, that we should not make any assumptions and perform this sanitization (normalization, actually) on a per-LLM provider basis at the least.

Also, this kind of impl. detail of LLM-specific requirements shouldn't leak into code that handles OpenAPI - one should perform this normalization just-in-time, i.e, just before passing it to the generator.

Reply:
Ok, I'll close this PR and rethink it now